rust/lance-encoding/src/encoder.rs

```diff
@@ -264,7 +284,9 @@ impl CoreArrayEncodingStrategy {
                 }
             }
             _ => Ok(Box::new(BasicEncoder::new(Box::new(
-                ValueEncoder::try_new(data_type, get_compression_scheme())?,
+                ValueEncoder::try_new(Arc::new(CoreBufferEncodingStrategy {
+                    compression_scheme: get_compression_scheme(),
+                }))?,
```

Comment:
Could the buffer encoding strategy be a property of `CoreArrayEncodingStrategy` instead of `ValueEncoder`?  In other words, could this be something like...

```
_ => {
  let buffer_encoder = self.buffer_encoding_strategy.create_buffer_encoder(...);
  Ok(Box::new(BasicEncoder::new(Box::new(ValueEncoder::try_new(buffer_encoder)?))))
}
```

It would require changing `array_encoder_from_type` (taking in a data type) to `create_array_encoder` (taking in a slice of array ref) but that change is fine and I think I end up making that change anyways in some pending PR.

Reply:
Yeah that would be straight forward.

We also instantiate ValueEncoder here for the `ListOffsetEncoder`:
https://github.com/lancedb/lance/blob/da28952d47af2ac58c58e6f9aaca3f3c06476771/rust/lance-encoding/src/encodings/logical/list.rs#L831-L844

Do you think it makes sense to pass the BufferEncodingStrategy through to this constructor from the constructor of `ListFieldEncoder` here? (e.g., make the BufferEncodingStrategy a property of `ListOffsetEncoder`)
https://github.com/lancedb/lance/blob/da28952d47af2ac58c58e6f9aaca3f3c06476771/rust/lance-encoding/src/encodings/logical/list.rs#L1092-L1098

Then we could create the inner `ValueEncoder` inline when in `make_encode_task`?
https://github.com/lancedb/lance/blob/da28952d47af2ac58c58e6f9aaca3f3c06476771/rust/lance-encoding/src/encodings/logical/list.rs#L876-L877